File: Leetcode_Python/leaf-similar-trees.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def leafSimilar(self, root1, root2):
        """
        :type root1: Optional[TreeNode]
        :type root2: Optional[TreeNode]
        :rtype: bool
        """
        
        root1Vals = []
        root2Vals = []

        def getValues(root, rootVals):
            if not root:
                return

            if not root.left and not root.right:
                rootVals.append(root.val)

            getValues(root.left, rootVals)
            getValues(root.right, rootVals)

        getValues(root1, root1Vals)
        getValues(root2, root2Vals)

        return str(root1Vals) == str(root2Vals)

File: Leetcode_JS/leaf-similar-trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */
var leafSimilar = function(root1, root2) {
    let root1Vals = []
    let root2Vals = []
    const collectLeafValues = (root, rootVals) => {
        if (!root) return
        if (!root.left && !root.right) rootVals.push(root.val)

        collectLeafValues(root.left, rootVals)
        collectLeafValues(root.right, rootVals)
    }

    collectLeafValues(root1, root1Vals)
    collectLeafValues(root2, root2Vals)
    
    return root1Vals.toString() == root2Vals.toString()
};